daemon/cluster/services.go

```diff
@@ -260,13 +263,51 @@ func (c *Cluster) ServiceLogs(ctx context.Context, input string, config *backend
 		stdStreams = append(stdStreams, swarmapi.LogStreamStderr)
 	}
 
+	// Get tail value squared away - the number of previous log lines we look at
+	var tail int64
+	if config.Tail == "all" {
+		// tail of 0 means send all logs on the swarmkit side
+		tail = 0
+	} else {
+		t, err := strconv.Atoi(config.Tail)
+		if err != nil || t < 0 {
+			// TODO(dperny) better error message
+			return errors.New("tail relative to start of logs not supported")
```

Comment:
My suggestion:

```go
if err != nil {
        return errors.New(`Tail must be a non-negative integer, or "all"`)
}
if t < 0 {
        return errors.New("negative Tail values are not supported")
}
```

(I don't actually think the current log message is bad for the second case, but I have a slight preference for talking about the domain of the parameter, instead of what a negative parameter would do if it was supported.)